Week4/app/src/main/java/com/em4n0101/mymovies/MoviesAdapter.kt

```diff
@@ -0,0 +1,30 @@
+package com.em4n0101.mymovies
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.em4n0101.mymovies.data.Movie
+import com.em4n0101.mymovies.utils.MovieViewHolder
+
+class MoviesAdapter(private val movieList: ArrayList<Movie>, val delegate: SelectItemListener): RecyclerView.Adapter<MovieViewHolder>() {
```

Comment:
Don't pass the data through the constructor. Create a separate method for that. 

```kotlin
fun setData(movies: List<Movie>) {
this.movies.clear()
this.movies.addAll(movies)
notifyDataSetChanged()
}
```

I wouldn't call this Listener parameter `delegate`. `clickListener` is more appropriate.